aodncore/pipeline/steps/harvest.py

```diff
@@ -403,5 +411,125 @@ def run_additions(self, harvester_map, tmp_base_dir):
                     self.storage_broker.upload(pipeline_files=files_to_upload)
 
 
+class CsvHarvesterRunner(BaseHarvesterRunner):
+    """:py:class:`BaseHarvesterRunner` implementation to load csv pipeline files to the database."""
+
+    def __init__(self, storage_broker, harvest_params, config, logger):
+        super().__init__(config, logger)
+        if harvest_params is None:
+            harvest_params = {}
+
+        self.params = harvest_params
+        self.storage_broker = storage_broker
+        self.config = self._config
+        self.logger = self._logger
+        self.pipeline_files = None
+
+    def run(self, pipeline_files):
+        """The entry point to the generic csv harvester
+
+        :return: None
+        """
+        validate_pipelinefilecollection(pipeline_files)
+
+        self.pipeline_files = pipeline_files
+        objs = self.params.get('db_objects')
+        if objs:
+            runsheet = (x for x in map(self.build_runsheet, objs) if x)
+        else:
+            raise MissingConfigParameterError('Generic CSV Harvester requires that the '
+                                              'harvest_params["db_objects"] attribute be set')
+
+        with DatabaseInteractions(config=self.get_config_file('database.json'),
+                                  schema_base_path=self._config.pipeline_config['harvester']['schema_base_dir'],
+                                  logger=self.logger) as conn:
+            process = self.get_process_sequence(conn)
+            for step in runsheet:
+                self._logger.info('Executing steps for {}'.format(step['name']))
+                for task in process:
+                    getattr(conn, task)(step)
+
+        files_to_upload = pipeline_files.filter_by_bool_attribute('pending_store_addition')
+        if files_to_upload:
+            self.storage_broker.upload(pipeline_files=files_to_upload)
+
+        metadata = self.params.get('metadata_updates')
+        if metadata and len(metadata) > 0:
+            gn_config = self.get_config_file('metadata.json')
+            gn = Geonetwork(
+                base_url=gn_config['metadata_url'],
+                username=gn_config['metadata_username'],
+                password=gn_config['metadata_password']
+            )
+            with DatabaseInteractions(config=self.get_config_file('database.json'),
+                                      schema_base_path=self._config.pipeline_config['harvester']['schema_base_dir'],
+                                      logger=self.logger) as conn:
+                for m in metadata:
+                    if 'spatial' in m:
+                        m['spatial']['db_schema'] = self.params['db_schema']
+                    handler = GeonetworkMetadataHandler(conn, gn, m, self.logger)
+                    handler.run()
```

Comment:
@mhidas re:

> Only one issue with it: if this part fails, it currently raises an unhandled exception, so the pipeline ends up in an ERROR state. This is despite having successfully harvested the data. I think the desired behaviour in such a case would be to only issue a warning, but allow the pipeline to end with a SUCCESS status.

There are a number of potential exceptions in the classes called in this section of _CsvHarvesterRunner.run()_, so we could probably move it to the end of the method and embed it in a try statement - catch any exception and just log it a warning

Reply:
Yep, that would be good. In fact we could be a bit picky about which exceptions we catch. In particular, I think anything to do with talking to Geonetwork could be caught.
Exceptions arising from querying the database could be allowed through, as they might indicate a config error or something more sinister.

Reply:
so more like...

```python
            except [InvalidSQLConnectionError, InvalidSQLTransactionError, MissingFileError]:
                raise
            except Exception as e:
                self._logger.warning(e)
```

Reply:
I've created [fix-geonetwork-error-fail](https://github.com/aodn/python-aodncore/tree/fix-geonetwork-error-fail) as a fork of [csv-harvester](https://github.com/aodn/python-aodncore/tree/csv-harvester) and committed the above change to that branch - but I won't be able to test it before I go on leave...